discord-scripts/invite-management.ts

```diff
@@ -133,10 +209,86 @@ export default async function sendInvite(discordClient: Client, robot: Robot) {
           }
         } catch (error) {
           robot.logger.error(
-            `An error occurred setting up the defense audit channel: ${error}`,
+            `An error occurred setting up the defense ${auditChannelType.toLowerCase()} audit channel: ${error}`,
           )
         }
       }
     })
+
+    // WIP, just testing out invite counting in order to verify which invite was used on join
+    discordClient.on("guildMemberAdd", async (member: GuildMember) => {
+      // for debugging
+      robot.logger.info(member)
+
+      const oldInvites =
+        (guildInvites.get(member.guild.id) as Collection<
+          string,
+          { uses: number }
+        >) || new Collection<string, { uses: number }>()
+      const fetchedInvites = await member.guild.invites.fetch()
+      const newInvites = new Collection<string, number>(
+        fetchedInvites.map((invite) => [invite.code, invite.uses ?? 0]),
+      )
+      guildInvites.set(member.guild.id, newInvites)
+
+      robot.logger.info(
+        `Old Invites: ${JSON.stringify(Array.from(oldInvites.entries()))}`,
+      )
+      robot.logger.info(
+        `New Invites: ${JSON.stringify(Array.from(newInvites.entries()))}`,
+      )
+
+      const usedInvite = fetchedInvites.find((fetchedInvite) => {
+        const oldInvite = oldInvites.get(fetchedInvite.code)
+        const oldUses =
+          typeof oldInvite === "object" ? oldInvite.uses : oldInvite
+        return (fetchedInvite.uses ?? 0) > (oldUses ?? 0)
+      })
+
+      robot.logger.info(`Used Invite: ${usedInvite ? usedInvite.code : "None"}`)
+
+      if (usedInvite && usedInvite.channelId) {
+        const channel = member.guild.channels.cache.get(
+          usedInvite.channelId,
+        ) as TextChannel
+        if (channel) {
+          const channelTypeMatch = channel.name.match(/(ext|int)-(.*)-audit/)
+          const clientName = channelTypeMatch
+            ? channelTypeMatch[2]
+                .replace(/-/g, " ")
+                .split(" ")
+                .map(
```

Comment:
Going to refactor this around to just match ChannelID -> ChannelName -> Assigned roles and then grab role with `Defense External: Client` rather than falling overmyself with these consts 😄